scripts/update_from_schema.js

```diff
@@ -0,0 +1,372 @@
+}
+
+function getTypeOfField(field_value) {
+    let type = field_value["type"];
+
+    if (type == "array") {
+        type = getTypeOfField(field_value["items"]);
+    }
+
+    return type;
+}
+
+function getExampleValues(key, field_value, host_number) {
+    example = _.get(field_value, "example");
+
+    if (example == undefined) {
+        throw `ERROR: string field ${key} missing example values`;
+    }
+
+    values = example.split(",");
+    values = values.map(s => s.trim());
+
+    value = values[host_number]
+    if (value == "null") {
+        value = null
+    }
+
+    return value;
+}
+
+function getItemsIfArray(field_value) {
+    if (_.has(field_value, "items")) {
+        field_value = field_value["items"];
+    }
+
+    return field_value;
+}
+
+function generateSystemProfileValues(field, host_number) {
+    let value = {}
+
+    if (_.has(field, "items")) {
+        field = field["items"];
+    }
+
+    for (let [key, field_value] of Object.entries(field["properties"])) {
+        let type = getTypeOfField(field_value);
+
+        field_value = getItemsIfArray(field_value);
+
+        //for strings take example values from example property
+        if (type == "object") {
+            //console.log(`Found object: ${key} | value: ${field_value}`)
+            value[key] = generateSystemProfileValues(field_value, host_number);
+        } else if (type == "string") {
+            ///console.log(field_value);
+            value[key] = getExampleValues(key, field_value, host_number);
+        } else if (type == "boolean") {
+            value[key] = Boolean(host_number);
+        } else if (type == "integer") {
+            value[key] = host_number;
+        } else {
+            throw `ERROR! ${key} type: ${type} not supported!`;
+        }
+    }
+
+    return value
+}
+
+function generateNewSystemProfileFacts(schema, number_of_hosts) {
+    let new_host_system_profile_facts = [];
+
+    for (let i = 0; i < number_of_hosts; i++) {
+        new_host_system_profile_facts.push(generateSystemProfileValues(schema, i));
+    }
+
+    return new_host_system_profile_facts;
+}
+
+function updateHostsJson(new_host_system_profile_facts) {
+    console.log("\n### updating example data for tests in hosts.json ###");
+
+    let hosts = getHosts();
+
+    let i = 0;
+    _.forEach(hosts, (host) => {
+        if (host["account"] == "test") {
+            _.forEach(new_host_system_profile_facts[i], (value, field) => {
+                host["system_profile_facts"][field] = new_host_system_profile_facts[i][field];
+            })
+        }
+        i++;
+    })
+
+    fs.writeFileSync(HOSTS_FILE_PATH, JSON.stringify(hosts, null, 4));
+}
+
+function getTestHostIds(hosts) {
+    let test_host_ids = [];
+
+    _.forEach(hosts, (host) => {
+        if (host["account"] == "test") {
+            test_host_ids.push(host["id"]);
+        }
+    })
+
+    return test_host_ids;
+}
+
+function updateHostTests(schema, new_host_system_profile_facts) {
```

Comment:
VSCode shows `updateHostTests` and `new_host_system_profile_facts` are defined but not used, is they really not used or get used behind the scenes.  i.e. not providing them shows errors at execution time.

Reply:
Draft PR. I've fixed this in the current local version. Some leftovers from a different approach to dynamic test creation